crypto/ocsp/ocsp_test.cc

```diff
@@ -0,0 +1,216 @@
+    0xeb, 0x47, 0xda, 0xdb, 0x26, 0xa6, 0x99, 0x58, 0x3d, 0xcc, 0xa3, 0x37,
+    0x41, 0xd8, 0xe4, 0xb8, 0x88, 0x4a, 0x55, 0xb5, 0x8e, 0xfc, 0x95, 0xed,
+    0x78, 0xca, 0xcd, 0x29, 0x87, 0x0d, 0xdb, 0x09, 0x2f, 0x57, 0x7e, 0x41,
+    0x94, 0x85, 0x8d, 0x13, 0xd7, 0xa2, 0x7b, 0xf8, 0xaa, 0x4f, 0x4f, 0xe4,
+    0x0a, 0x52, 0x04, 0xc5, 0xeb, 0x65, 0x63, 0x04, 0x43, 0xee, 0xe2, 0x9a,
+    0x6a, 0xf8, 0xaa, 0xb0, 0xc4, 0xc4, 0xd3, 0xf6, 0x4b, 0xbd, 0x70, 0xca,
+    0x57, 0x89, 0xbd, 0x98, 0x8b, 0x9a, 0xb4, 0xb5, 0xdb, 0x94, 0x6d, 0x52,
+    0xa0, 0x2a, 0x03, 0x19, 0x61, 0xee, 0xb8, 0x03, 0xe7, 0x79, 0x9d, 0xb7,
+    0x52, 0xb7, 0x87, 0xc0, 0xc6, 0xc7, 0xec, 0xf8, 0x91, 0xc6, 0xc1, 0x87,
+    0xf7, 0x6e, 0xb8, 0x13, 0xd5, 0x74, 0x3f, 0x48, 0xb1, 0xcd, 0x19, 0x75,
+    0xcb, 0xb1, 0x34, 0x4a, 0x23, 0x07, 0xb3, 0x3e, 0xd1, 0x13, 0x74, 0x14,
+    0x7a, 0xf3, 0x80, 0x15, 0x9e, 0x87, 0x2d, 0x41, 0xef, 0x77, 0x0e, 0x58,
+    0x0c, 0xcf, 0xf8, 0xe3, 0x59, 0xce, 0xc3, 0xb6, 0xe4, 0x7b, 0x24, 0x47,
+    0x9a, 0x1a, 0xb6, 0x8c, 0xd4, 0xa9, 0x7f, 0x5c, 0x35, 0x74, 0x78, 0xf0,
+    0xa5, 0x70, 0xdc, 0x17, 0xe7, 0x4a, 0xec, 0x92, 0x54, 0x92, 0x64, 0xd8,
+    0xbd, 0xf8, 0x32, 0x2a, 0x07, 0x32, 0x90, 0x44, 0xd1, 0xdc, 0x91, 0x1f,
+    0x79, 0x68, 0xea, 0x88, 0x92, 0xc7, 0xb0, 0xc6, 0x1c, 0xed, 0x29, 0xc7,
+    0x68, 0x7f, 0x06, 0x43, 0x33, 0x25, 0x05, 0xd1, 0xe5, 0x2a, 0xb4, 0xd9,
+    0xfe, 0xc3, 0x01, 0x8d, 0x6a, 0x4e, 0x27, 0x49, 0xc4, 0x93, 0xe9, 0x0f,
+    0x5a, 0xc0, 0x20, 0x55, 0xe8, 0xa2, 0xbc, 0x42, 0x53, 0x5d, 0x45, 0x33,
+    0xb3, 0x08, 0xc6, 0xc6, 0x17, 0x2d, 0x5c, 0x19, 0x09, 0x13, 0xfd, 0x1f,
+    0x24, 0xe9, 0x3d, 0x48, 0xff, 0x50, 0x81, 0x50, 0xcc, 0x80, 0x42, 0xf3,
+    0x77, 0xb0, 0xff, 0xa9, 0xe7, 0x60, 0x70, 0x97, 0x5d, 0xb8, 0xcb, 0x78,
+    0x15, 0x50, 0x3b, 0x8b, 0x0a, 0x9a, 0x9a, 0xb0, 0xbe, 0x61, 0x5f, 0x0e,
+    0xfa, 0x1e, 0xd9, 0x10, 0x26, 0xb8, 0x03, 0xff, 0x50, 0x2b, 0x6c, 0x06,
+    0x56, 0xaf, 0x5e, 0xa4, 0x74, 0x51, 0xac, 0x7a, 0x6f, 0x81, 0x63, 0xf2,
+    0xd2, 0x74, 0xdd, 0x1f, 0x9d, 0x17, 0xc8, 0x90, 0x9e, 0xa4, 0x79, 0xc5,
+    0x5f, 0xd7, 0x12, 0xb2, 0x39, 0x9c, 0x06, 0x35, 0x45, 0x9a, 0xae, 0xa4,
+    0x14, 0x17, 0x67, 0xd7, 0x9f, 0xa2, 0x41, 0x86, 0x4d, 0xd4, 0x31, 0xd5,
+    0x1a, 0xbe, 0x8e, 0x5c, 0xa9, 0x0d, 0xdc, 0x8a, 0x0a, 0xa9, 0x65, 0x4a,
+    0xd5, 0x38, 0x22, 0xc8, 0x14, 0x5b, 0xb9, 0x5f, 0x50, 0x8b, 0x94 };
+
+
+static bssl::UniquePtr<OCSP_RESPONSE> LoadOCSP_RESPONSE(bssl::Span<const uint8_t> der) {
+  const uint8_t *ptr = der.data();
+  return bssl::UniquePtr<OCSP_RESPONSE>(d2i_OCSP_RESPONSE(nullptr, &ptr, der.size()));
+}
+
+TEST(OCSPTest, TestBasic) {
+  bssl::UniquePtr<OCSP_RESPONSE> ocsp_response;
+  bssl::UniquePtr<OCSP_BASICRESP> basic_response;
+
+  ocsp_response = LoadOCSP_RESPONSE(ocsp_response_der);
+  ASSERT_TRUE(ocsp_response);
+
+  int ocsp_status = OCSP_response_status(ocsp_response.get());
+  ASSERT_EQ(OCSP_RESPONSE_STATUS_SUCCESSFUL, ocsp_status);
```

Comment:
I see 6 different `OCSP_RESPONSE_*` values. Can we return all those?

Reply:
Unfortunately, the current unit tests from s2n only test upon OCSP der responses that guarantee to return OCSP_RESPONSE_STATUS_SUCCESSFUL. 
I was hoping to add further robust testing beyond s2n's scope as a separate PR after all the initial s2n support work was done, since creating and verifying the new test files will take some time.

Reply:
clarified offline: we got plans to add more tests :)